### DISCUSSION TITLE
Shared library

### DISCUSSION THREAD

**Discussion by bsrona** [jrouwe/JoltPhysics#442] [Q&A] [answered]
Can I make shared library projects with Cmake? Is the cross platform determinism affected by other module if I link execute by static library?

**Comment by jrouwe** [comment, marked as answer]
Jolt cannot be used as a shared library as it doesn't mark any classes/functions for export. There are however a couple of projects that wrap Jolt into a C API that can be used to create a shared library (see: main page, [Bindings for other Languages](https://github.com/jrouwe/JoltPhysics#bindings-for-other-languages), C). For maximum performance I would recommend statically linking the library.

Edit: A shared library version is available since d2f1d97004d036c6f759203c42e264e401472037.

I don't understand your second question. The rules for a deterministic simulation are outlined in the [API documentation](https://jrouwe.github.io/JoltPhysics/), I don't think it matters what other modules you link with unless those are not deterministic of course.

**Reply by bsrona** [reply]
The second question means that I don't know if some linker options affect the cross platform determinism. I don't think it matters too.
